registry/reference.go

```diff
@@ -164,13 +175,26 @@ func (r Reference) ValidateReference() error {
 	if r.Reference == "" {
 		return nil
 	}
-	if _, err := r.Digest(); err == nil {
+
+	return fmt.Errorf("%w: invalid non-tag, non-digest reference: %v", errdef.ErrInvalidReference, r.Reference)
+}
+
+// ValidateReferenceAsTag validates the reference as a tag.
+func (r Reference) ValidateReferenceAsTag() error {
+	if tagRegexp.MatchString(r.Reference) {
 		return nil
 	}
-	if !tagRegexp.MatchString(r.Reference) {
-		return fmt.Errorf("%w: invalid tag", errdef.ErrInvalidReference)
+
+	return fmt.Errorf("%w: invalid tag", errdef.ErrInvalidReference)
+}
+
+// ValidateReferenceAsDigest validates the reference as a digest.
+func (r Reference) ValidateReferenceAsDigest() (err error) {
+	if _, err = r.Digest(); err == nil {
+		return nil
 	}
-	return nil
+
+	return fmt.Errorf("%w: invalid digest; %v", errdef.ErrInvalidReference, err)
```

Comment:
This wasn't a type; it's of the general form:

`<errorClass>: <description>`, where description contains usual grammatical syntax, which includes the semicolon (join ideas that are related and equal in grammatical structure; such as this very sentence :)

Reply:
Do we have a doc for `;`? If yes, please provide a link for that. Usually, `:` means a chain of errors where `;` means multiple unrelated errors.

Reply:
Oh I see, you're saying that in the context of errors, `;` already has a special meaning.  I wasn't aware of that; will change the error so that it doesn't rely in a semicolon.

Reply:
Also the wrapping here was unnecessary; now just `return registry.Reference{}, err`.  Fixed.